### PR TITLE
added documentation about the audio and voice infrastructure

### PR DESCRIPTION
**Pull request by kaikreuzer**
Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: docs/documentation/concepts/audio.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+# Audio & Voice
+
+Audio and voice features are an important aspect of any smart home solution as it is a very natural way to interact with the user.
+
+Eclipse SmartHome comes with a very modular architecture that enables all kinds of different use cases. At its core, there is the notion of an *audio stream*. Audio streams are provided by *audio sources* and consumed by *audio sinks*.  
+
+![](images/audio.png)
+
+- *Audio Streams* are essentially a byte stream with a given *audio format*. They can do not need to be limited in size, i.e. it is also allowed to have continuous streams, e.g. the input from a microphone or from an Internet radio station.
```
Comment (BClark09) [inline docs/documentation/concepts/audio.md]: Remove can in "They can do not"

File: docs/documentation/concepts/audio.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+- *Speech-to-Text* (STT) services are similar to audio sinks, but they do not simply play back the stream, but convert it to a plain string. They provide information about supported formats and locales.
+
+As plain text from an STT service is often not very useful, there is additionally the concept of a *human language interpreter*:
+
+![](images/hli.png)
+
+A *Human Language Interpreter* takes a string as an input. It then derives actions from it (like sending commands to devices) and/or replies with a string, which opens the possibility to realizes conversations. As such an interpreter is not directly linked to audio streams, but operates on strings only, this can be the basis for any kind of assistant, e.g. for chat bots using the console, XMPP, Twitter or other messaging services.
```
Comment (BClark09) [inline docs/documentation/concepts/audio.md]: Change Realizes to realize in "possibility to realizes conversations."

File: docs/documentation/concepts/audio.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+- *Speech-to-Text* (STT) services are similar to audio sinks, but they do not simply play back the stream, but convert it to a plain string. They provide information about supported formats and locales.
+
+As plain text from an STT service is often not very useful, there is additionally the concept of a *human language interpreter*:
+
+![](images/hli.png)
+
+A *Human Language Interpreter* takes a string as an input. It then derives actions from it (like sending commands to devices) and/or replies with a string, which opens the possibility to realizes conversations. As such an interpreter is not directly linked to audio streams, but operates on strings only, this can be the basis for any kind of assistant, e.g. for chat bots using the console, XMPP, Twitter or other messaging services. 
+
+Applications can dynamically chose which services to use, so that different sinks can be used for different use cases. Defaults can be set as a configuration for all those services in case an application does not ask for any specific service.
```
Comment (BClark09) [inline docs/documentation/concepts/audio.md]: Change chose to choose in "can dynamically chose"

File: docs/documentation/features/index.md [hunk 4]
```diff
@@ -47,15 +47,24 @@ Installation and uninstallation requests are executed by a thread pool named "ex
...
 
-Bindings can optionally include [discovery services](../concepts/discovery.html), which allow the system to automatically find accessible devices and services.
+Bindings can optionally include [discovery services](../concepts/discovery.html), which allow the system to automatically find accessible devices and services. Furthermore, they can register devices as [audio sources and audio sinks](../concepts/audio.html), so that microphones and speakers can be made available to the audio and voice support features of the framework.
 
 ### User Interfaces
 
 User interfaces normally use the REST API for communication, but if they are not client-side, but served from the runtime, they also have the option to use all local Java services.
 
-Currently, the only available user interface in Eclipse SmartHome is the Classic UI, but further UIs will be available soon.
+Currently, there are 3 available user interface in Eclipse SmartHome: the Classic UI, the Basic UI and the Paper UI.
```
Comment (BClark09) [inline docs/documentation/features/index.md]: Change interface to interfaces in "3 available user interfaces"

File: docs/documentation/features/index.md [hunk 5]
```diff
@@ -47,15 +47,24 @@ Installation and uninstallation requests are executed by a thread pool named "ex
...
 All user interfaces can share icon sets, so that these do not have to be included in every single user interface.
 Eclipse SmartHome comes with the following iconsets:
 
  - `org.eclipse.smarthome.ui.iconset.classic`: [Classic Icon Set](ui/iconset/classic/readme.html)
+
+### Voice Services
+ 
+Voice extensions provide implementations for Text-to-Speech, Speech-to-Text and Human Language Interpreter services.
+ 
+These services are often very solution specific, so that is no one-fits-all implementation in Eclipse SmartHome.
```
Comment (BClark09) [inline docs/documentation/features/index.md]: Chat that to there in "so that is no one-fits-all"

**Comment by kaikreuzer** [comment]
@BClark09 I definitely do not mind, thanks for noticing all my typos! It seems I am not concentrated enough today to write documentation :-) Should be all fixed now!

**Comment by BClark09** [comment]
No worries! It all looks fine and reads well to me now!

**Comment by kaikreuzer** [comment]
rebased the branch to be mergeable again.

**Comment by kaikreuzer** [comment]
@maggu2810 Ok to merge this as well?

**Pull request merged**